scripts/run.sh

```diff
@@ -8,31 +19,142 @@ function depends {
 
-download=false
+  docker-compose -f docker-compose-production.yml up ${is_build} --detach
+}
+
+stop_prod() {
+  if [ ! -f "docker-compose-production.yml" ]; then 
+    echo "Error: docker-compose-production.yml missing, which is required for docker-compose"
+    echo "Please check the repo for missing files"
+    exit 1
+  fi
+
+  echo "Stopping containers in production mode"
+
+  docker-compose -f docker-compose-production.yml down
+}
+
+run_certbot() {
+  echo
+  echo "Will generate a brand new cert"
+  echo
+  docker exec sdschedule-certbot sed -i -r 's/(listen .*443)/\1;#/g; s/(ssl_(certificate|certificate_key|trusted_certificate) )/#;#\1/g' /etc/nginx/sites-enabled/sdschedule.conf
+  docker exec sdschedule-web nginx -s reload
+  docker exec sdschedule-certbot certbot certonly --webroot -w /var/www/certbot -d sdschedule.com -n --force-renewal --agree-tos --email "$letsencrypt_email"
+  docker exec sdschedule-certbot sed -i -r 's/#?;#//g' /etc/nginx/sites-enabled/sdschedule.conf
+  docker exec sdschedule-web nginx -s reload
+}
+
+main() {
+  check_directory
+  check_depend
+
+  case $environment in
+    PROD)
+      run_prod
+    ;;
+    DEV)
+      run_dev
+    ;;
+    *)
+      echo "Internal Error: Invalid Environment Setting"
+      exit -1
+    ;;
+  esac
+}
+
+if [ $# -eq 0 ]; then
+  echo "Run in development mode without downloading and rebuilding by default"
+  echo "Try '-h' or '--help' for help message"
+  echo
+fi
 
 while test $# -gt 0; do
-   case "$1" in 
-       -h | --help) echo "-h, --help        Gives the possible commands"
-          echo "-d, --download    Will download data fresh from Schedule of Classes"
+  case "$1" in 
+      -h | --help) 
+          echo "The run script for the UCSD Schedule Planner" 
+          echo ""
+          echo "-h, --help         Show this very helpful message"
+          echo "-d, --download     Will download data fresh from Schedule of Classes"
+          echo "-p, --production   Will run in production mode (detached)"
+          echo "-b, --build        Rebuild the services. Do this if files are modified"
+          echo "-s, --stop         Stop the detached *production* services"
+          echo "-c, --cert <email> Run certbot for production service for first time (after started). Use an email address as argument"
+          echo ""
+          echo "Sample usage: "
+          echo ""
+          echo "-p -d -b          Rebuild the services, download fresh data, and run in production mode"
+          echo "-p                Run in production mode. Will build services if and only if no services built previously"
           exit 0
           ;;
       -d | --download)
-          download=true
-          shift
+          SDSCHEDULE_SCRAPE=1
+          ;;
+      -p | --production)
+          environment=${ENV_PROD}
+          ;;
+      -b | --build)
+          is_build=${BUILD}
+          ;;
+      -s | --stop)
+          stop_prod # move to new location?
```

Comment:
what's this comment amount?

Reply:
This function call is a little bit awkward. Might come up with more graceful ways to do it.